src/server/pps/server/master.go

```diff
@@ -334,13 +301,10 @@ func (a *apiServer) upsertWorkersForPipeline(pipelineInfo *pps.PipelineInfo) err
 		workerRc, err := rc.Get(
 			ppsutil.PipelineRcName(pipelineInfo.Pipeline.Name, pipelineInfo.Version),
 			metav1.GetOptions{})
-		if err == nil {
-			if (workerRc.Spec.Template.Spec.Containers[0].Resources.Requests == nil) && *workerRc.Spec.Replicas == 1 {
-				parallelism = 1
-				resourceRequests = nil
-				resourceLimits = nil
-			}
+		if err != nil {
```

Comment:
Shouldn't we be returning the error if there is an error here? I noticed you have a todo below to find out why the statement below is running when there is an error.

Reply:
Good point, I'm going to change it and see if CI passes. I'd be very surprised if it didn't since I'm pretty sure if we got an error there it would cause a segfault as is, so we must not be getting errors there.